File: backend/dto/src/main/java/pl/dawid0604/pcForum/dto/user/UserProfileDetailsDTO.java
package pl.dawid0604.pcForum.dto.user;

import java.util.List;

public record UserProfileDetailsDTO(String encryptedId, String avatar, String nickname, String rank, String createdAt, String lastActivity,
                                    long numberOfPosts, long numberOfThreads, long numberOfUpVotes, long numberOfDownVotes,
                                    long numberOfVisits, long numberOfObservations, List<ActivityDTO> activities, boolean isLoggedUser,
                                    boolean isOnline, boolean isObserved) {

    public record ActivityDTO(String title, String description, String date, String encryptedId, ActivityType type) { }

    public enum ActivityType { FOLLOW, CREATED_POST, COMMENTED_POST, VOTED_UP,
                               VOTED_DOWN, CREATED_THREAD, COMMENTED_THREAD }
}

File: frontend/src/app/model/UserProfileDetailsDTO.ts
export interface UserProfileDetailsDTO {
    encryptedId: string,
    avatar: string,
    nickname: string,
    rank: string,
    numberOfPosts: number,
    numberOfThreads: number,
    numberOfUpVotes: number,
    numberOfDownVotes: number,
    numberOfVisits: number,
    numberOfObservations: number,
    createdAt: string,
    lastActivity: string,
    activities: ActivityDTO[],
    isLoggedUser: boolean,
    isOnline: boolean
    isObserved: boolean
}

export interface ActivityDTO {
    title: string,
    description: string,
    date: string,
    encryptedId: string,
    type: ActivityType
}

export enum ActivityType {
    FOLLOW, CREATED_POST, COMMENTED_POST, VOTED_UP,
    VOTED_DOWN, CREATED_THREAD, COMMENTED_THREAD
}